src/lib/tls/tls12/msg_client_kex.cpp

```diff
@@ -22,6 +23,25 @@
 
 namespace Botan::TLS {
 
+bool is_valid_ec_shared_secret(const Group_Params named_group, const secure_vector<uint8_t>& shared_secret)
+   {
+      // RFC 8422 - 5.11.
+      //   With X25519 and X448, a receiving party MUST check whether the
+      //   computed premaster secret is the all-zero value and abort the
+      //   handshake if so, as described in Section 6 of [RFC7748].
+      if(named_group == Group_Params::X25519/* || named_group == Group_Params::X448 (not implemented)*/)
+         {
+         // Side channel resistent (!!!) check for 0
+         uint8_t or_sum = 0;
+         for(uint8_t s_byte : shared_secret)
+            or_sum |= s_byte;
+
+         return (or_sum != 0);
```

Comment:
For this loop use the const time utilities in `ct_utils.h`.

Reply:
Added `CT::all_zeros` in #3522 to make this easy

Reply:
Nice, thanks!